File: app_agents/triage_agent.py

from agents import Agent

from .admission_agent import admission_agent
from .catalog_agent import catalog_agent
from .info_agent import info_agent
from .order_agent import order_agent
from .party_agent import party_agent

triage_agent = Agent(
    name="ToyShopTriageAgent",
    instructions=(
        "Greet every caller with: 'Welcome to kids for fun Poughkeepsie Galleria Mall: 2001 South Rd Unit A108, Poughkeepsie, NY.' "
        "Then ask how you can help and briefly list what you can do: store info, toy catalog help, "
        "admission tickets and policies, birthday party planning, and order management (purchases, status, payments, refunds). "
        "After the greeting, decide the need and hand off to the best-fit specialist agent."
    ),
    handoffs=[info_agent, catalog_agent, admission_agent, party_agent, order_agent],
)

File: src/agents/triage.agent.js
/**
 * Triage Agent - Main entry point that routes to specialist agents
 */

import { Agent } from '@openai/agents';
import { infoAgent } from './info.agent.js';
import { catalogAgent } from './catalog.agent.js';
import { admissionAgent } from './admission.agent.js';
import { partyAgent } from './party.agent.js';
import { orderAgent } from './order.agent.js';

const AGENT_STYLE = `
You are a friendly Kids4Fun voice assistant at Playfunia.

Guidelines:
- Talk like a real phone agent with 1-2 short sentences per response.
- After answering, wait for the caller to respond.
- Only provide extra details if asked.
- Be warm, helpful, and concise.
`.trim();

export const triageAgent = Agent.create({
    name: 'TriageAgent',
    instructions: `${AGENT_STYLE}

GREETING (use on first contact):
"Welcome to Kids for Fun at Poughkeepsie Galleria Mall, 2001 South Road Unit A108, Poughkeepsie, New York. How can I help you today? I can assist with store information, toy catalog, admission tickets, birthday party planning, or order management."

You are the main receptionist. Your job is to:
1. Greet callers warmly
2. Understand their needs quickly
3. Hand off to the right specialist agent

ROUTING GUIDE:
- Store info, FAQs, policies, locations, hours, staff → InfoAgent
- Toys, products, catalog, inventory → CatalogAgent  
- Admission tickets, waivers, check-in, entry → AdmissionAgent
- Birthday parties, bookings, packages, party rooms → PartyAgent
- Orders, payments, refunds, promotions → OrderAgent

RULES:
- If the request is outside these domains, say: "I can only help with store info, toys, admissions, parties, or orders."
- Don't try to answer specialist questions yourself - hand off quickly
- Be polite and efficient
- Confirm the handoff: "Let me connect you with our [specialist] who can help with that."`,
    handoffs: [infoAgent, catalogAgent, admissionAgent, partyAgent, orderAgent],
});

// Export all agents for use elsewhere
export { infoAgent, catalogAgent, admissionAgent, partyAgent, orderAgent };
